### PR TITLE
Add docs for i wallet provider

### PR DESCRIPTION
**Pull request by akifuji**

### CODE REVIEW

File: src/web3bch-providers/IWalletProvider.ts [hunk 1]
```diff
@@ -3,47 +3,196 @@ import Utxo from "./entities/Utxo"
...
+   */
   getAddressIndex(
     changeType: ChangeType,
     dAppId?: string
   ): Promise<number>
 
+  /**
+   * Returns the stored redeem scripts related to the DApp ID.
+   * @example
+   * const redeemScripts = await provider.getRedeemScript(
```
Comment (usatie) [inline src/web3bch-providers/IWalletProvider.ts]: typo : `getRedeemScript`
-> `getRedeemScripts`

File: src/web3bch-providers/IWalletProvider.ts [hunk 2]
```diff
@@ -3,47 +3,196 @@ import Utxo from "./entities/Utxo"
...
+   *   "53212266f7994100e442f6dff10fbdb50a93121d25c196ce0597517d35d42e68"
+   * )
+   * console.log(redeemScript)
```
Comment (usatie) [inline src/web3bch-providers/IWalletProvider.ts]: typo : `console.log(redeemScript)`
-> `console.log(redeemScripts)`

File: src/web3bch-providers/IWalletProvider.ts [hunk 3]
```diff
@@ -3,47 +3,196 @@ import Utxo from "./entities/Utxo"
...
+   * > ["03424f587e06424954424f5887", "789787a72c21452a1c98ff"]
+   * @param dAppId The DApp ID. If no dAppId is set the default DApp ID will be set.
+   * @returns The stored redeem script list.
```
Comment (usatie) [inline src/web3bch-providers/IWalletProvider.ts]: The list of redeem scripts stored.

File: src/web3bch-providers/INetworkProvider.ts [hunk 4]
```diff
@@ -1,10 +1,45 @@
...
+   * @returns The protocol version.
+   */
   getProtocolVersion(): Promise<number>
 
+  /**
+   * Returns the current network.
+   * @example
+   * const network = await provider.getNetwork()
+   * console.log(network)
+   * > e3e1f3e8
```
Comment (Yuiki) [inline src/web3bch-providers/INetworkProvider.ts]: -> 3823236072 (decimal)

File: src/web3bch-providers/IWalletProvider.ts [hunk 5]
```diff
@@ -3,47 +3,196 @@ import Utxo from "./entities/Utxo"
...
+   * @returns The protocol version.
+   */
   getProtocolVersion(): Promise<number>
 
+  /**
+   * Returns the current network.
+   * @example
+   * const network = await provider.getNetwork()
+   * console.log(network)
+   * > e3e1f3e8
```
Comment (Yuiki) [inline src/web3bch-providers/IWalletProvider.ts]: -> 3823236072

**Pull request merged**